includes/utils/class-amp-dom-utils.php

```diff
@@ -175,26 +159,7 @@ static function( $noscript_matches ) {
 			return false;
 		}
 
-		// Remove pre-HTML5-style encoding declaration if added above.
-		if ( $added_back_compat_meta_content_type ) {
-			$meta_http_equiv_element = $dom->getElementById( 'meta-http-equiv-content-type' );
-			if ( $meta_http_equiv_element ) {
-				$meta_http_equiv_element->parentNode->removeChild( $meta_http_equiv_element );
-			}
-		}
-
-		// Make sure there is a head and a body.
-		$head = $dom->getElementsByTagName( 'head' )->item( 0 );
-		if ( ! $head ) {
-			$head = $dom->createElement( 'head' );
-			$dom->documentElement->insertBefore( $head, $dom->documentElement->firstChild );
-		}
-		$body = $dom->getElementsByTagName( 'body' )->item( 0 );
-		if ( ! $body ) {
-			$body = $dom->createElement( 'body' );
-			$dom->documentElement->appendChild( $body );
-		}
```

Comment:
Is the ensuring of the `body` and `head` being done now?

Reply:
The `AMP_DOM_Document` now enforces a base structure:
https://github.com/ampproject/amp-wp/blob/7513e4145c393d699cede7ec32fd4c037a38835b/includes/utils/class-amp-dom-document.php#L170-L188